src/harness/reference_models/iap/iap.py

```diff
@@ -0,0 +1,623 @@
+            num_unsatisfied_grants_channel[index] )
+        else:
+          fairshare_channel.append(0)
+
+      # List of grants initialized with max EIRP
+      grants_eirp = [grant.max_eirp for grant in neighborhood_grants]
+      
+      # Initialize list of grants_satisfied to False
+      grants_satisfied = [False] * num_unsatisfied_grants
+      
+      while num_unsatisfied_grants > 0:
+
+        for g_idx, grant in enumerate(neighborhood_grants):
+
+          if grants_satisfied[g_idx] is False:
+            grant_good_over_all_channels = True
+
+            # Set to True if grant frequency range overlaps with a channel, 
+            # otherwise, set to False
+            grants_overlap_flag = [False] * len(channels)
+
+            # Initialize interference per channel list 
+            interference_list = [0] * len(channels) 
+
+            for idx, channel in enumerate(channels):
+              # check if grant interferes with protection point, channel
+              grant_overlap_check = interf.grantFrequencyOverlapCheck(grant, 
+                                           channel[0], channel[1], protection_ent_type)
+          
+              # Update True or False for a particular channel index
+              grants_overlap_flag[idx] = grant_overlap_check 
+              # if grant overlaps with protection point over channel 
+              if grant_overlap_check:
+                # Compute interference grant causes to protection point over channel
+                interference = interf.dbToLinear(interf.computeInterference(grant, 
+                                 grants_eirp[g_idx], channel_constraint, fss_info, 
+                                 esc_antenna_info, region_type))
+                # if calculated interference is more than fair share of
+                # interference to which the grants are entitled then 
+                # grant is considered unsatisfied
+                if interference > fairshare_channel[idx]:
+                  grant_good_over_all_channels = False
+                  break
+                else:
+                  interference_list[idx] = interference
+       
+          if grant_good_over_all_channels is True:
+            grants_satisfied[g_idx] = True
+            if num_unsatisfied_grants > 0:
+              num_unsatisfied_grants = num_unsatisfied_grants - 1
+            for ch_idx, g_ch_flag in enumerate(grants_overlap_flag):
+              # Grant interferes with protection point over channel 
+              if g_ch_flag:
+                iap_threshold_channel[ch_idx] = iap_threshold_channel[ch_idx] - \
+                                                  interference_list[ch_idx]
+                num_unsatisfied_grants_channel[ch_idx] = num_unsatisfied_grants_channel[ch_idx] - 1
+                # Re-calculate fairshare for a channel
+                if num_unsatisfied_grants_channel[ch_idx] > 0:
+                  fairshare_channel[ch_idx] = float (iap_threshold_channel[ch_idx]) /  \
```

Comment:
According to spec, I think you need to go over all neighborhood_grants first to see whether they are satisfied. Only after that, you can update fairshare_channel. In other words, the code should look something like:

```
while num_unsatisfied_grants > 0:
  for g_idx, grant in enumerate(neighborhood_grants):
    go over all overlapping 5mhz channel to see if grant_good_over_all_channels
  for index, channel in enumerate(channels):
    update fairshare_channel.
```

Reply:
Updated   the logic to run over all the grants for all the channels before updating   fairshare